### PR TITLE
Merge with PostgreSQL 9.5 (up to almost 9.6devel is stamped)

### PR DESCRIPTION
**Pull request by pengzhout**
Notable upstream changes:

    * Allow INSERTs that would generate constraint conflicts to be turned
      into UPDATEs or ignored. The syntax is INSERT ... ON CONFLICT DO
      NOTHING/UPDATE. This is the Postgres implementation of the popular
      UPSERT command. Append-Only table and ORCA are not yet supported.

    * Add row-level security control so tables can have row security
      policies that restrict, on a per-user basis, which rows can be
      returned by normal queries or inserted, updated, or deleted by data
      modification commands.

    * Add Block Range Indexes, BRIN indexes store only summary data
      (such as minimum and maximum values) for ranges of heap blocks.
      Append-Only table and ORCA are not yet supported.

    * Substantial performance improvement. Improve the speed of sorting of
      varchar, text, and numeric fields via "abbreviated" keys; Improve lock
      scalability, this particularly addresses scalability problems when
      running on systems with multiple CPU sockets; Improve performance of
      hash joins and bitmap index scan, etc.

    * Allow setting multiple target columns in an UPDATE from the result of
      a single sub-SELECT. This is accomplished using the syntax UPDATE tab
      SET (col1, col2, ...) = (SELECT ...).

    * Add SELECT option TABLESAMPLE to return a subset of a table.
      Append-Only table and ORCA are not yet supported.

    * Simplify WAL record format, this allows external tools to more easily
      track what blocks are modified.

    Notable GPDB changes:

    * GROUPING SETS is no longer MPP-aware and not supported by hashagg.

      GPDB implement GROUPING SETS with a number of issues, ranging from
      assertion failures to incorrect results and now PG9.5 introduced its
      own implementation of GROUPING SETS, CUBE, ROLLUP, we decided to
      replace it with the one of PG9.5. As discussed in
      https://groups.google.com/a/greenplum.org/d/topic/gpdb-dev/z9Ww2lU5_fE/discussion
      it's not MPP-aware and needs a gather node to bring all data to the
      QD node and perform the grouping there, meanwhile, it only supports
      sorted aggregates now. The following commits will address those two
      problems.

    Co-authored-by: Asim R P <apraveen@pivotal.io>
    Co-authored-by: Chris Hajas <chajas@pivotal.io>
    Co-authored-by: David Kimura <dkimura@pivotal.io>
    Co-authored-by: Georgios Kokolatos <gkokolatos@pivotal.io>
    Co-authored-by: Heikki Linnakangas <hlinnakangas@pivotal.io>
    Co-authored-by: Jinbao Chen <jinchen@pivotal.io>
    Co-authored-by: Jimmy Yih <jyih@pivotal.io>
    Co-authored-by: Pengzhou Tang <ptang@pivotal.io>

### CODE REVIEW

File: concourse/pipelines/gpdb_opensource_release.yml [hunk 1]
```diff
@@ -0,0 +1,292 @@
+resource_types:
+- name: gcs
+  type: docker-image
```
Comment (asimrp) [inline concourse/pipelines/gpdb_opensource_release.yml]: `gpdb_opensource_release.yml` should not be added by this PR.  It was moved from master to another repo by commit  8490aaa30cb9dee2b0e631de4dae146f8d23db72.

**Comment by hlinnaka** [comment]
Looks good at a quick glance. Thanks to everyone involved!

I'm glad we got the new WAL record format now. That simplifies pg_rewind a lot, and also removed the annoying diff we've been carrying in all our redo functions, where the arguments where slightly different from upstream's.

**Comment by ashwinstar** [comment]
On Tue, Aug 27, 2019 at 12:29 PM Heikki Linnakangas <
notifications@github.com> wrote:

> Looks good at a quick glance. Thanks to everyone involved!
>
> I'm glad we got the new WAL record format now. That simplifies pg_rewind a
> lot, and also removed the annoying diff we've been carrying in all our redo
> functions, where the arguments where slightly different from upstream's.
>
+1

**Comment by yydzero** [comment]
Great achievement!

File: concourse/pipelines/gpdb_opensource_release.yml (hunk 1 above)
Comment (pengzhout) [reply concourse/pipelines/gpdb_opensource_release.yml]: Thanks, will remove it.

**Comment by pengzhout** [comment]
> I'm glad we got the new WAL record format now. That simplifies pg_rewind a lot, and also removed the annoying diff we've been carrying in all our redo functions, where the arguments where slightly different from upstream's.

Agreed, that's a very impressive change in the 9.5 merge, thanks for the great work.

**Comment by guofengrichard** [comment]
I'm most excited that we aligned the implementation of grouping sets with upstream.  That helps us get rid of lots of issues and makes the future life much easier.

**Comment by soumyadeep2007** [comment]
We need to address this issue before we merge in the 9_5 merge PR into gpdb master? 
https://github.com/greenplum-db/gpdb-postgres-merge/issues/47

**Comment by pengzhout** [comment]
Merged. Thanks for the reviews.

**Pull request merged**